tests/ztest/include/ztest_test.h

```diff
@@ -18,6 +18,7 @@ struct unit_test {
 	void (*test)(void);
 	void (*setup)(void);
 	void (*teardown)(void);
+	u32_t thread_options;
```

Comment:
unit_test strcut looks generic fo far and by adding thread_options we are adding test specific stuff, i think if we can avoid 
this it would be clean.

Reply:
> info related specifically to the test only

The mode that the test thread runs in (user vs supervisor) is part of the configuration of the specific test case, I do not see a better place to put it.

Reply:
I agree that this is only struct we are passing at this stage.
If the need comes in future to add more variables related to test infra vs actual test then we should look at segregating them.

I am fine with current changes. Thanks.